OpenScienceJournal/whistlepunk_library/src/main/java/com/google/android/apps/forscience/ble/MkrSciBleManager.java

```diff
@@ -0,0 +1,372 @@
+                                l.onValuesUpdated(values);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static double[] parse(ValueType valueType, byte[] value) {
+        if (ValueType.UINT8.equals(valueType)) {
+            if (value.length < 1) {
+                return null;
+            }
+            final ByteBuffer buffer = ByteBuffer.allocate(4);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put(value[0]);
+            buffer.position(0);
+            return new double[]{buffer.getInt()};
+        }
+        if (ValueType.UINT16.equals(valueType)) {
+            if (value.length < 2) {
+                return null;
+            }
+            final ByteBuffer buffer = ByteBuffer.allocate(4);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put(value[1]);
+            buffer.put(value[0]);
+            buffer.position(0);
+            return new double[]{buffer.getInt()};
+        }
+        if (ValueType.UINT32.equals(valueType)) {
+            if (value.length < 4) {
+                return null;
+            }
+            final ByteBuffer buffer = ByteBuffer.allocate(8);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put((byte) 0);
+            buffer.put(value[3]);
+            buffer.put(value[2]);
+            buffer.put(value[1]);
+            buffer.put(value[0]);
+            buffer.position(0);
+            return new double[]{buffer.getLong()};
+        }
+        if (ValueType.SFLOAT.equals(valueType)) {
+            if (value.length < 4) {
+                return null;
+            }
+            final ByteBuffer buffer = ByteBuffer.allocate(4);
+            buffer.put(value[3]);
+            buffer.put(value[2]);
+            buffer.put(value[1]);
+            buffer.put(value[0]);
+            buffer.position(0);
+            return new double[]{buffer.getFloat()};
+        }
+        if (ValueType.SFLOAT_ARR.equals(valueType)) {
+            final int size = value.length / 4;
+            final double[] array = new double[size];
+            final ByteBuffer buffer = ByteBuffer.allocate(4);
+            int c = 0;
+            for (int i = 0; i < size; i++) {
+                final int offset = 4 * i;
+                buffer.position(0);
+                buffer.put(value[3 + offset]);
+                buffer.put(value[2 + offset]);
+                buffer.put(value[1 + offset]);
+                buffer.put(value[offset]);
+                buffer.position(0);
+                array[i] = buffer.getFloat();
+            }
+            return array;
+        }
+        return null;
+    }
+
+    private enum ValueType {
+        UINT8, UINT16, UINT32, SFLOAT, SFLOAT_ARR
+    }
+
+    public interface Listener {
```

Comment:
Can you add a javadoc comment for this interface?

Reply:
Done.